MSW mock generation produces multipleOf: undefined for integer types, causing TypeScript errors with exactOptionalPropertyTypes: true

## Title
`multipleOf: undefined` generated in MSW mocks causes TypeScript error with `exactOptionalPropertyTypes: true`

## Description

**Affected versions:** v7.10.0 and later (confirmed in v7.11.2)

**Current behavior:**
When generating MSW mocks from OpenAPI schemas containing integer fields without a `multipleOf` constraint, Orval generates:
```javascript
faker.number.int({min: 0, max: 9999, multipleOf: undefined})
```

This causes TypeScript compilation errors when `exactOptionalPropertyTypes: true` is enabled:
```
error TS2345: Argument of type '{ min: number; max: number; multipleOf: undefined; }' is not assignable to parameter of type 'FakerNumberIntOptions<number>'
```

**Expected behavior:**
The `multipleOf` parameter should be omitted when not defined in the OpenAPI schema:
```javascript
faker.number.int({min: 0, max: 9999})
```

**Root cause:**
In [`packages/mock/src/faker/getters/scalar.ts#L23`](https://github.com/orval-labs/orval/blob/e70256718431c40a323e28ac4e56675fa4033190/packages/mock/src/faker/getters/scalar.ts#L23):
```typescript
${isFakerV9 ? `, multipleOf: ${item.multipleOf}` : ''}
```
The code doesn't check if `item.multipleOf` is undefined before interpolating it.

**Proposed solution:**
```typescript
${isFakerV9 && item.multipleOf !== undefined ? `, multipleOf: ${item.multipleOf}` : ''}
```

## Steps to reproduce
1. Create an OpenAPI schema with integer fields (without `multipleOf` property)
   ```yaml
   type: integer
   minimum: 0
   maximum: 9999
   ```
2. Set `exactOptionalPropertyTypes: true` in tsconfig.json
3. Run Orval to generate MSW mocks
4. Run TypeScript compiler

## Environment
- **Orval:** 7.11.2
- **TypeScript:** 5.7.3
- **@faker-js/faker:** 10.0.0
- **Node.js:** 20.x

## Impact
This prevents projects using strict TypeScript settings from upgrading to Orval v7.10.0 or later.

## Workaround
Currently, there's no clean workaround without either:
- Disabling `exactOptionalPropertyTypes` (impacts entire project)
- Downgrading to Orval v7.9.0 (missing new features)
- Post-processing generated files (not maintainable)

## Additional context
This issue was introduced in v7.10.0 when adding support for Faker.js v9's `multipleOf` option.

--- a/packages/mock/src/faker/getters/scalar.test.ts
+++ b/packages/mock/src/faker/getters/scalar.test.ts
@@ -149,6 +149,121 @@ describe('getMockScalar (example handling with falsy values)', () => {
   });
 });
 
+describe('getMockScalar (multipleOf handling)', () => {
+  const createContext = (
+    packageJsonDeps?: Record<string, string>,
+  ): ContextSpecs => {
+    const context = {
+      output: {
+        override: {},
+        ...(packageJsonDeps && {
+          packageJson: { dependencies: packageJsonDeps },
+        }),
+      },
+    } as ContextSpecs;
+    return context;
+  };
+
+  const baseArg = {
+    imports: [],
+    operationId: 'test-operation',
+    tags: [],
+    existingReferencedProperties: [],
+    splitMockImplementations: [],
+  };
+
+  it('should include multipleOf when defined for integer type with Faker v9', () => {
+    const integerType: SchemaObjectType = 'integer';
+    const result = getMockScalar({
+      ...baseArg,
+      item: {
+        type: integerType,
+        minimum: 0,
+        maximum: 100,
+        multipleOf: 5,
+        name: 'test-item',
+      },
+      context: createContext({ '@faker-js/faker': '^9.0.0' }),
+    });
+
+    expect(result.value).toBe(
+      'faker.number.int({min: 0, max: 100, multipleOf: 5})',
+    );
+  });
+
+  it('should not include multipleOf when undefined for integer type with Faker v9', () => {
+    const integerType: SchemaObjectType = 'integer';
+    const result = getMockScalar({
+      ...baseArg,
+      item: {
+        type: integerType,
+        minimum: 0,
+        maximum: 100,
+        multipleOf: undefined,
+        name: 'test-item',
+      },
+      context: createContext({ '@faker-js/faker': '^9.0.0' }),
+    });
+
+    expect(result.value).toBe('faker.number.int({min: 0, max: 100})');
+  });
+
+  it('should not include multipleOf for integer type with Faker v8', () => {
+    const integerType: SchemaObjectType = 'integer';
+    const result = getMockScalar({
+      ...baseArg,
+      item: {
+        type: integerType,
+        minimum: 0,
+        maximum: 100,
+        multipleOf: 5,
+        name: 'test-item',
+      },
+      context: createContext({ '@faker-js/faker': '^8.0.0' }),
+    });
+
+    expect(result.value).toBe('faker.number.int({min: 0, max: 100})');
+  });
+
+  it('should handle multipleOf for number (float) type', () => {
+    const numberType: SchemaObjectType = 'number';
+    const result = getMockScalar({
+      ...baseArg,
+      item: {
+        type: numberType,
+        minimum: 0,
+        maximum: 100,
+        multipleOf: 0.5,
+        name: 'test-item',
+      },
+      context: createContext(),
+    });
+
+    expect(result.value).toBe(
+      'faker.number.float({min: 0, max: 100, multipleOf: 0.5})',
+    );
+  });
+
+  it('should use fractionDigits when multipleOf is undefined for number type', () => {
+    const numberType: SchemaObjectType = 'number';
+    const result = getMockScalar({
+      ...baseArg,
+      item: {
+        type: numberType,
+        minimum: 0,
+        maximum: 100,
+        name: 'test-item',
+      },
+      mockOptions: { fractionDigits: 2 },
+      context: createContext(),
+    });
+
+    expect(result.value).toBe(
+      'faker.number.float({min: 0, max: 100, fractionDigits: 2})',
+    );
+  });
+});
+
 describe('getMockScalar (nested arrays handling)', () => {
   it('should generate valid syntax for nested arrays (array of arrays)', () => {
     const result = getMockScalar({

--- a/packages/mock/src/faker/getters/scalar.ts
+++ b/packages/mock/src/faker/getters/scalar.ts
@@ -133,7 +133,7 @@ export const getMockScalar = ({
           ? 'bigInt'
           : 'int';
       let value = getNullable(
-        `faker.number.${intFunction}({min: ${item.minimum ?? mockOptions?.numberMin}, max: ${item.maximum ?? mockOptions?.numberMax}${isFakerV9 ? `, multipleOf: ${item.multipleOf}` : ''}})`,
+        `faker.number.${intFunction}({min: ${item.minimum ?? mockOptions?.numberMin}, max: ${item.maximum ?? mockOptions?.numberMax}${isFakerV9 && item.multipleOf !== undefined ? `, multipleOf: ${item.multipleOf}` : ''}})`,
         item.nullable,
       );
       if (type === 'number') {

